packages/devtools_app/lib/src/debugger/debugger_controller.dart

```diff
@@ -1018,10 +1018,22 @@ class DebuggerController extends DisposableController
     }
   }
 
-  void toggleFileOpener(bool visible) {
+  void toggleFileOpenerVisibility(bool visible) {
     _showFileOpener.value = visible;
   }
 
+  void hideSearchInFileIfVisible() {
+    if (showSearchInFileField.value) {
+      toggleSearchInFileVisibility(false);
+    }
+  }
+
+  void hideFileOpenerIfVisible() {
+    if (showFileOpener.value) {
```

Comment:
ValueNotifier will not notify listeners if the new value is the same as the current value, so there is no need to add extra methods `hideSearchInFileIfVisible` and `hideFileOpenerIfVisible` to check the current state of the respective value notifiers. You can just call the `toggleSearchInFileVisibility` and `toggleFileOpenerVisibility` directly where you are calling these hide methods

Reply:
Ah good to know, thanks!